iree/compiler/Conversion/LinalgToLLVM/LinalgBufferizePass.cpp

```diff
@@ -172,6 +173,53 @@ LogicalResult convertAnyLinalgOp(OpBuilder &b, linalg::LinalgOp op,
   return success();
 }
 
+static LogicalResult convertTransferOp(OpBuilder &b,
+                                       VectorTransferOpInterface op,
+                                       BlockAndValueMapping &bvm) {
+  if (op.getShapedType().isa<MemRefType>()) return success();
+  assert(op->getNumResults() == 1);
+  OpBuilder::InsertionGuard g(b);
+  b.setInsertionPoint(op);
+  Location loc = op.getLoc();
+  Value newInputBuffer = bvm.lookup(op.source());
+  if (auto tensorType =
+          op->getResult(0).getType().dyn_cast<RankedTensorType>()) {
+    // If the op return a Tensor allocate a buffer for the returned value.
+    auto tensorShape = tensorType.getShape();
+    auto memrefType = MemRefType::get(tensorShape, tensorType.getElementType());
+    SmallVector<Value, 4> dynOperands;
+    Value outputTensor = op->getResult(0);
+    for (auto dim : llvm::enumerate(tensorShape)) {
+      if (dim.value() == TensorType::kDynamicSize) {
```

Comment:
please refactor so the loop checks `tensorType.isDynamicDim(idx)` instead of leaking the underlying constant.